client/src/components/Jobs/Jobs.jsx

```diff
@@ -133,15 +134,25 @@ function JobsView({
     {
       Header: 'Job Name',
       accessor: 'name',
-      Cell: (el) => (
-        <span
-          title={el.value}
-          id={`jobID${el.row.index}`}
-          className="job__name"
-        >
-          {el.value}
-        </span>
-      ),
+      Cell: (el) => {
+        const query = queryStringParser.parse(useLocation().search);
+        return (
+          <span
+            title={el.value}
+            id={`jobID${el.row.index}`}
+            className="job__name"
+          >
+            {query.query_string ? (
+              <HighlightSearchTerm
+                searchTerm={query.query_string}
+                content={el.value}
+              />
+            ) : (
+              el.value
+            )}
```

Comment:
How about changing `<HighlightSearchTerm>` to internally handle the repeat `query.query_string ?` condition via `searchTerm`, then removing the repeat `query.query_string ? (…) : ( … )}` code?
```suggestion
            <HighlightSearchTerm
              searchTerm={query.query_string}
              content={el.value}
            />
```

Reply:
Good suggestion, @wesleyboar. However, this conditional rendering method is more suitable for cases like these since it doesn't need to render the additional component `HighlightSearchTerm` when no search term is present thereby improving performance. When user has a long job history, it might lead to more overhead as HighlightSearchTerm component will be rendered even when search query isn't provided if I make this change. Let me know your thoughts and I can address that.

Reply:
Okay, cool. Thanks for explaining. Performance is not my strong suit. I trust you guys.

Reply:
No worries. I might be wrong so if @chandra-tacc or @shayanaijaz can weigh in.

Reply:
I think this is good as is. It is also more readable this way since it's clear when search terms are highlighted and when they are not